Dockerfile.service

```diff
@@ -1,11 +1,45 @@
-# you probably don't want to use this Dockerfile directly,
+# you probably don't want to build this Dockerfile directly,
 # use `make <svc>` instead
 
+#
+# pull deps
+#
+FROM debian:12.8 AS deps
+
+
+# renovate: datasource=github-releases depName=golang-migrate/migrate
+ARG MIGRATE_VERSION=4.18.1
+
+
+RUN apt update && \
+	apt install -y wget
+
+# get migrate
+RUN wget -q https://github.com/golang-migrate/migrate/releases/download/v$MIGRATE_VERSION/migrate.linux-amd64.tar.gz -O - | tar -xzvf - migrate
+
+
 # builder builds a go service
 # the build args SERVICE and VERSION must be set!
 
 FROM golang:1.23-bookworm AS builder
 
+#
+# build spice helper (for migrations)
+#
+
+WORKDIR /build/spice
+
+COPY go.* ./
+COPY libs libs
+COPY spicedb/migrations spicedb/migrations
+RUN go mod download
+COPY cmd/spice cmd/spice
+RUN CGO_ENABLED=0 GOOS=linux go build -a -o spice cmd/spice/spice.go
```

Comment:
I guess summarize this real quick. One `COPY` one `RUN`.

Reply:
pretty sure this was for dependency caching reasons